standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2360,6 +2360,20 @@ public Table getTable(String catName, String dbName, String tableName, String va
     return deepCopyTables(FilterUtils.filterTablesIfEnabled(isClientFilterEnabled, filterHook, tabs));
   }
 
+  @Override
+  public List<Table> getTableObjectsByRequest(GetTablesRequest req) throws TException {
```

Comment:
Do you need to do any special handling of SessionMetastoreClient?

Reply:
You are referring to SessionHiveMetaStoreClient right ?

SessionHiveMetaStoreClient does not have an implementation for get_partitions_with_specs and piggybacks on the implementation in its superclass (HiveMetaStoreClient) from the inheritance hierarchy. I just followed the same pattern here.

Also it just returns a list of table objects, basically a read query and should work the same across sessions, since, it just returns persisted session information.